#!/usr/bin/env python
PACKAGE = "mbzirc_husky"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("tolerance", double_t, 0, "Ransac tolerance",   0.06 , 0,  3)
gen.add("angleTolerance", double_t, 0, "Angle Ransac tolerance",   0.08 , 0,  4)
gen.add("distanceTolerance", double_t, 0, "Distance tolerance",   0.08 , 0,  2)
gen.add("distance", double_t, 0, "Distance",   0.4 , 0,  10)
gen.add("fwSpeed", double_t, 0, "Forward Speed",   0.1 , 0,  10)
gen.add("minPoints", int_t, 0, "Minimum points",   0 , 0,  100)
gen.add("maxEval", int_t, 0, "Maximal evaluation Points",   0 , 0,  1000)
gen.add("spdLimit", double_t, 0, "Speed Limit",   0.3 , 0,  10)
gen.add("realConst", double_t, 0, "Z coordinate correction",   1 , 0,  1)
gen.add("dispConst", double_t, 0, "Displacement correction",   3 , 0,  3)
gen.add("covarianceBricks", int_t, 0, "Covariance Limit for stop exploration",   500 , 0,  10000)
gen.add("covariancePattern", int_t, 0, "Covariance Limit for stop exploration",  1500 , 0,  10000)


exit(gen.generate(PACKAGE, "brick_pile", "brick_pile"))
